src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+    func_8003E398(&colCtx->polyNodes);
+    func_8003E3AC(globalCtx, &colCtx->polyNodes, tblMax, colCtx->colHeader->nbPolygons);
 
-u32 func_80041D94(u32 a0, u32 a1, u32 a2) {
-    u32 uVar1;
+    lookupTblMemSize = func_8003BB18(colCtx, globalCtx, colCtx->lookupTbl);
+    osSyncPrintf(VT_FGCOL(GREEN));
+    osSyncPrintf("/*---結局 BG使用サイズ %dbyte---*/\n", memSize + lookupTblMemSize);
+    osSyncPrintf(VT_RST);
 
-    uVar1 = func_800419B0(a0, a1, a2, 0);
-    return uVar1 >> 0x15 & 0x1f;
+    func_8003E954(globalCtx, &colCtx->dyna);
+    func_8003E9A0(globalCtx, &colCtx->dyna);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041DB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041DE4.s")
+// Get CollisionHeader
+// original name: T_BGCheck_getBGDataInfo
+CollisionHeader* func_8003C4C4(CollisionContext* colCtx, s32 bgId) {
+    if (bgId == BG_ACTOR_MAX) {
+        return colCtx->colHeader;
+    }
+    if (bgId < 0 || bgId > BG_ACTOR_MAX) {
+        return NULL;
+    }
+    if (!(colCtx->dyna.bgActorFlags[bgId] & 1)) {
+        osSyncPrintf(VT_COL(YELLOW, BLACK));
+        osSyncPrintf("T_BGCheck_getBGDataInfo():そのbg_actor_indexは使われておりません。index=%d\n");
+        osSyncPrintf(VT_RST);
+        return NULL;
+    }
+    return colCtx->dyna.bgActors[bgId].colHeader;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041E18.s")
+// Test if pos is near collision boundaries
+s32 func_8003C55C(CollisionContext* colCtx, Vec3f* pos) {
+    if (pos->x < (colCtx->minBounds.x - 50.0f) || (colCtx->maxBounds.x + 50.0f) < pos->x ||
+        pos->y < (colCtx->minBounds.y - 50.0f) || (colCtx->maxBounds.y + 50.0f) < pos->y ||
+        pos->z < (colCtx->minBounds.z - 50.0f) || (colCtx->maxBounds.z + 50.0f) < pos->z) {
+        return false;
+    }
+    return true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041E4C.s")
+// Raycast Toward Floor
+// returns the yIntersect of the nearest poly found directly below `pos`, or BGCHECK_Y_MIN if no floor detected
+// returns the poly found in `outPoly`, and the bgId of the entity in `outBgId`
+f32 func_8003C614(GlobalContext* globalCtx, CollisionContext* colCtx, u16 xpFlags, CollisionPoly** outPoly,
+                  s32* outBgId, Vec3f* pos, Actor* actor, u32 arg7, f32 chkDist) {
+
+    f32 yIntersectDyna;
+    s32* temp_a0;
+    Lookup* lookupTbl;
+    Vec3f checkPos;
+    Lookup* lookup;
+    DynaRaycast dynaRaycast;
+    f32 yIntersect;
+
+    *outBgId = BGCHECK_SCENE;
+    *outPoly = NULL;
+    lookupTbl = colCtx->lookupTbl;
+    yIntersect = BGCHECK_Y_MIN;
+    checkPos = *pos;
+
+    while (true) {
+        if (checkPos.y < colCtx->minBounds.y) {
+            break;
+        }
+        if (BgCheck_PosErrorCheck(&checkPos, "../z_bgcheck.c", 4410)) {
+            if (actor != NULL) {
+                osSyncPrintf("こいつ,pself_actor->name %d\n", actor->id);
+            }
+        }
+        lookup = func_8003AD00(colCtx, lookupTbl, &checkPos);
```

Comment:
```suggestion
        subdivLookup = func_8003AD00(colCtx, lookupTbl, &checkPos);
```
just to be clearer, renaming functions would make this irrelevant